oc-includes/osclass/classes/controller/admin/CAdminUsers.php

```diff
@@ -645,6 +649,24 @@ public function doModel()
                 osc_add_flash_ok_message($msg, 'admin');
                 $this->redirectTo(osc_admin_base_url(true) . '?page=users&action=ban');
                 break;
+            case ('user_login'):
+                osc_csrf_check();
+                $aUser = $this->userManager->findByPrimaryKey(Params::getParam('id'));
+                if (!count($aUser)) {
+                    osc_add_flash_error_message(_m("The user doesn't exist"), 'admin');
+                    $this->redirectTo(osc_admin_base_url(true) . '?page=users');
+                }
+
+                Session::newInstance()->_set('userId', $aUser['pk_i_id']);
+                Session::newInstance()->_set('userName', $aUser['s_name']);
+                Session::newInstance()->_set('userEmail', $aUser['s_email']);
+                $phone = $aUser['s_phone_mobile'] ?: $aUser['s_phone_land'];
+                Session::newInstance()->_set('userPhone', $phone);
+
+                osc_run_hook('after_login', $user, osc_user_dashboard_url());
+                osc_add_flash_ok_message(sprintf(_m('Logged in as %s successfully'), $aUser['s_name']));
+                $this->redirectTo(osc_apply_filter('correct_login_url_redirect', osc_user_dashboard_url()));
```

Comment:
No need for an added filter, it will be merged.

Reply:
No need to change anything?